### ISSUE TITLE
When attempting to create aws_launch_configuration with incorrect AMI, timeout instead of more useful error

### ISSUE THREAD

**Issue by sheeley** [hashicorp/terraform#4926]
I'd set an ami incorrectly on my `aws_launch_configuration`, and saw this output a few times this morning:
```
* aws_launch_configuration.lumen-ui: Error creating launch configuration: timeout while waiting for state to become '[success]'
* aws_launch_configuration.lumen-api: Error creating launch configuration: timeout while waiting for state to become '[success]'
```

I'm not familiar with the level of granularity the APIs provide when creating resources like this, but it'd be useful to verify things like AMI existence and provide more context (and perhaps avoid the timeout error entirely).

**Comment by emdupp** [comment]
Issue happens with incorrect data for other fields as well, not just image_id.

**Comment by diroussel** [comment]
I'm seeing this on Terraform v0.6.10 too.

Here is the output from my TF_LOG=TRACE logs:

```
2016/02/02 13:28:39 [DEBUG] apply: aws_launch_configuration.uwlp-launchconf-nodeapp: executing Apply
  enable_monitoring:           "" => "1"
  image_id:                    "" => "ami-47a23a30"
  instance_type:               "" => "t2.micro"
  key_name:                    "" => "terraform_env_dev"
  name:                        "" => "uwlp-launchconf-bastion"
  root_block_device.#:         "" => "<computed>"
  security_groups.#:           "" => "1"
  security_groups.2309388835:  "" => "sg-50772134"
aws_launch_configuration.uwlp-launchconf-nodeapp: Creating...
  associate_public_ip_address: "" => "0"
  ebs_block_device.#:          "" => "<computed>"
  ebs_optimized:               "" => "<computed>"
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [DEBUG] autoscaling create launch configuration: {
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   EbsOptimized: false,
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   ImageId: "ami-47a23a30",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   InstanceMonitoring: {
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:     Enabled: true
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   },
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   InstanceType: "t2.micro",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   KeyName: "terraform_env_dev",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   LaunchConfigurationName: "uwlp-launchconf-bastion",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   SecurityGroups: ["sg-50772134"]
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: }
2016/02/02 13:28:39 [  enable_monitoring:           "" => "1"
DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [DEBUG] Waiting for state to become: [success]
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [TRACE] Waiting 500ms before next try
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [DEBUG] autoscaling create launch configuration: {
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   EbsOptimized: false,
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   ImageId: "ami-8847edfb",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   InstanceMonitoring: {
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:     Enabled: true
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   },
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   InstanceType: "t2.micro",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   KeyName: "terraform_env_dev",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   LaunchConfigurationName: "uwlp-launchconf-nodeapp",
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws:   SecurityGroups: ["sg-53772137"]
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: }
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [DEBUG] Waiting for state to become: [success]
2016/02/02 13:28:39 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:39 [TRACE] Waiting 500ms before next try
  image_id:                    "" => "ami-8847edfb"
  instance_type:               "" => "t2.micro"
  key_name:                    "" => "terraform_env_dev"
  name:                        "" => "uwlp-launchconf-nodeapp"
  root_block_device.#:         "" => "<computed>"
  security_groups.#:           "" => "1"
  security_groups.558313732:   "" => "sg-53772137"
2016/02/02 13:28:40 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:40 [TRACE] Waiting 500ms before next try
2016/02/02 13:28:40 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:40 [TRACE] Waiting 500ms before next try
2016/02/02 13:28:40 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:40 [TRACE] Waiting 500ms before next try
2016/02/02 13:28:40 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:40 [TRACE] Waiting 500ms before next try
2016/02/02 13:28:41 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:41 [TRACE] Waiting 800ms before next try
2016/02/02 13:28:41 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:41 [TRACE] Waiting 800ms before next try
2016/02/02 13:28:42 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:42 [TRACE] Waiting 1.6s before next try
2016/02/02 13:28:42 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:42 [TRACE] Waiting 1.6s before next try
2016/02/02 13:28:43 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:28:44 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:44 [TRACE] Waiting 3.2s before next try
2016/02/02 13:28:44 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:44 [TRACE] Waiting 3.2s before next try
2016/02/02 13:28:44 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:28:44 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:28:44 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:28:47 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:47 [TRACE] Waiting 6.4s before next try
2016/02/02 13:28:47 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:47 [TRACE] Waiting 6.4s before next try
2016/02/02 13:28:48 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:28:49 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:28:49 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:28:49 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:28:53 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:28:54 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:54 [TRACE] Waiting 10s before next try
2016/02/02 13:28:54 [DEBUG] terraform-provider-aws: 2016/02/02 13:28:54 [TRACE] Waiting 10s before next try
2016/02/02 13:28:54 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:28:54 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:28:54 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:28:58 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:28:59 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:28:59 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:28:59 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:29:03 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:29:04 [DEBUG] terraform-provider-aws: 2016/02/02 13:29:04 [TRACE] Waiting 10s before next try
2016/02/02 13:29:04 [DEBUG] terraform-provider-aws: 2016/02/02 13:29:04 [TRACE] Waiting 10s before next try
2016/02/02 13:29:04 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:29:04 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:29:04 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:29:08 [DEBUG] vertex provider.aws (close), waiting for: aws_autoscaling_group.uwlp-asg-bastion
2016/02/02 13:29:09 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, waiting for: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:29:09 [DEBUG] vertex root, waiting for: provider.aws (close)
2016/02/02 13:29:09 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-bastion, waiting for: aws_launch_configuration.uwlp-launchconf-bastion
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalIf
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteDiff
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalIf
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalApplyProvisioners
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalApplyPost
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalIf
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteDiff
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalIf
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalWriteState
2016/02/02 13:29:09 [DEBUG] root: eval: *terraform.EvalApplyPost
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-nodeapp: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-nodeapp: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-nodeapp: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-nodeapp: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [TRACE] Exiting eval tree: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:29:09 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, got dep: aws_launch_configuration.uwlp-launchconf-nodeapp
2016/02/02 13:29:09 [DEBUG] vertex aws_autoscaling_group.uwlp-asg-nodeapp, got dep: module.vpc.output.public_subnets
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-bastion: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-bastion: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-bastion: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_launch_configuration.uwlp-launchconf-bastion: Error creating launch configuration: timeout while waiting for state to become '[success]'
2016/02/02 13:29:09 [TRACE] Exiting eval tree: aws_launch_configuration.uwlp-launchconf-bastion

```

**Comment by fromonesrc** [comment]
Seeing this on 0.6.10 as well. Same problem with an incorrect SSH key reference.

**Comment by akramhussein** [comment]
In 0.6.10 I'm recieving a similar issue but for ECS Services:

**Comment by spanktar** [comment]
Seeing the same with 0.6.11 for launch configuration. Checking the above possibilities. Interesting to note that a second run of `terraform apply` completes with no errors.

**Comment by blakeblackshear** [comment]
Different reason, same issue. With 0.6.11, I get:
```
Error applying plan:

2 error(s) occurred:

* aws_launch_configuration.rethinkdb_server: Error creating launch configuration: timeout while waiting for state to become '[success]'
* aws_launch_configuration.elasticsearch_server: Error creating launch configuration: timeout while waiting for state to become '[success]'
```

Reverting to 0.6.3 gives me:
```
Error applying plan:

2 error(s) occurred:

* aws_launch_configuration.rethinkdb_server: Error creating launch configuration: LimitExceeded: Launch Configuration limit exceeded - You may only have 100 launch configurations
    status code: 400, request id: [9365c7b4-d74e-11e5-9f7f-ad4e7312f799]
* aws_launch_configuration.elasticsearch_server: Error creating launch configuration: LimitExceeded: Launch Configuration limit exceeded - You may only have 100 launch configurations
    status code: 400, request id: [933dcd11-d74e-11e5-ae96-4db29713879f]
```

**Comment by revett** [comment]
+1 

Happened today w/ `0.6.11` when I tried to create a `aws_launch_configuration`. It failed as the `key_name` stated did not exist.

**Comment by tspecht** [comment]
How were you guys able to Debug this and find out the real cause? I already checked that I'm using the latest AMI so this cannot be the case

**Comment by tspecht** [comment]
In addition, I see the launch configuration in my ec2 console?

**Comment by blakeblackshear** [comment]
@tspecht I rolled back to 0.6.3, and that gave me a more useful error message

**Comment by ernoaapa** [comment]
Yeah same problem with 0.6.12. Switching to 0.6.3 helped me to find out that actually had typo in `"${my-variable-name}"` (missing `var.` at beginning)

**Comment by errordeveloper** [comment]
Same error, but none of the above symptoms, it simply exists already in my case.

**Comment by baskaran-md** [comment]
In my case, I had a wrong security group id, and I got the same timeout error..

```
* aws_launch_configuration.lc_test: Error creating launch configuration: timeout while waiting for state to become '[success]'
```

**Comment by imacube** [comment]
@tspecht  Everything looked correct but didn't work. I discovered that I needed to have an entry to create an IAM instance profile. With out it the IAM role wasn't listed when I tried to manually launch an instance or create a launch configuration.

```
resource "aws_iam_instance_profile" "my_instance_profile" {
    name  = "${aws_iam_role.my_instance_role.name}"
    path  = "/"
    roles = ["${aws_iam_role.my_instance_role.name}"]
}
```

When I manually created the IAM role through the AWS Web Console the profile was automatically created with a name matching the role name.

As for debugging, you can try turning on debug log messages for terraform:
```
TF_LOG=DEBUG terraform apply
```

**Comment by jevonearth** [comment]
I have run into this while using the https://github.com/hashicorp/best-practices/ examples. I was getting the same error when the launch config already existed (created by tf) and after I manually removed the scaling groups and launch config.

Tested using terraform 0.6.12 via atlas.
```
2 error(s) occurred:

* aws_launch_configuration.green: Error creating launch configuration: timeout while waiting for state to become '[success]'
* aws_launch_configuration.blue: Error creating launch configuration: timeout while waiting for state to become '[success]'
```

**Comment by jmccarty3** [comment]
I ran into a similar issue as this. It appears that he newest updates are hiding 
```"errorCode": "ValidationException",``` That is returned by the api calls. I was able to find the actual error (not timeout) by checking the logs in CloudTrail.

**Comment by jevonearth** [comment]
Yes, after upgrading to terraform 0.6.13 my error messages changed to:

* aws_launch_configuration.blue: Error creating launch configuration: ValidationError: AMI cannot be described
	status code: 400, request id: 22d1492f-ec2e-11e5-b6dc-49477e4bcc65
* aws_launch_configuration.green: Error creating launch configuration: ValidationError: AMI cannot be described
	status code: 400, request id: 22d17069-ec2e-11e5-b921-a91214c844cc

which was all because I had a missing base image that packer and applications depended on, and in turn terraform.

So my report a couple days ago is a non-issue.

**Comment by leigh507** [comment]
Hmmm, I am running 0.6.13 and i still see the same issue. 
Error applying plan:

1 error(s) occurred:

* aws_launch_configuration.launch_config: Error creating launch configuration: timeout while waiting for state to become '[success]'

After updating the AMI var to be the AMI ID rather than a reference to a global var set in platform.tf it works.

**Comment by autotune** [comment]
Hey all,

Using @imacube debug suggestion I found out it's looking for the specific provider to be specified somewhere, so in my example:

2016/03/23 00:30:41 [DEBUG] vertex provider.aws (close), waiting for: aws_launch_configuration.apses1
2016/03/23 00:30:41 [DEBUG] vertex root, waiting for: provider.aws (close)


    brian$ head -n5 main.tf 
    # Configure AWS provider

    provider "aws" {
    region = "ap-southeast-1"
    }

Hopefully this helps someone else as well!

**Comment by phinze** [comment]
Hey folks, this timeout error masking should be addressed in v0.6.14 :+1:
